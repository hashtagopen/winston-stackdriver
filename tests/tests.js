const { join } = require('path')
const { Readable } = require('stream')
const { test } = require('@ianwalter/bff')
const execa = require('execa')

const winstonStackdriver = join(__dirname, '../')
const lineOne = JSON.stringify({
  timestamp: '2019-09-10T01:15:12.511Z',
  level: 'debug',
  message: 'Not attempting to create collection'
}) + '\n'
const lineTwo = JSON.stringify({
  timestamp: '2019-09-10T01:15:12.559Z',
  level: 'info',
  message: 'Applied validation schema to willy wonka'
}) + '\n'

test('severity is added to log entry', ({ expect }) => {
  return new Promise(resolve => {
    const stdin = new Readable({ read () {} })
    const cp = execa('node', [winstonStackdriver])
    let counter = 0
    cp.stdout.on('data', data => {
      expect(JSON.parse(data)).toMatchSnapshot()
      if (counter) {
        resolve()
      } else {
        counter++
      }
    })
    stdin.pipe(cp.stdin)
    stdin.push(lineOne)
    stdin.push(lineTwo)
    stdin.push(null) // Push null to close the stream.
  })
})
