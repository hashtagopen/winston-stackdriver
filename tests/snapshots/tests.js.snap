// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`severity is added to log entry 1`] = `
Object {
  "message": "Not attempting to create collection",
  "severity": "DEBUG",
  "timestamp": "2019-09-10T01:15:12.511Z",
}
`;

exports[`severity is added to log entry 2`] = `
Object {
  "message": "Applied validation schema to willy wonka",
  "severity": "INFO",
  "timestamp": "2019-09-10T01:15:12.559Z",
}
`;
